### PR TITLE
Extend and clarify graphical primitives / paints

### PR DESCRIPTION
**Pull request by drott**
Group primitives, add transforms, composites. Explain high level design to be mapped to 2d graphics library primitives.

### CODE REVIEW

**Comment by drott** [comment]
Preview available here: https://github.com/drott/colr-gradients-spec/blob/primitivesExtend/colr-gradients-spec.md

File: colr-gradients-spec.md [hunk 1]
```diff
@@ -105,20 +118,31 @@ to be exposed to end-users.
...
 
-## Color Line
+### Solid
+
+A solid paint fills the drawing region with a solid color specified by
+`ColorIndex`. `ColorIndex` references color `paletteIndex` from the `CPAL`
+palette, and applies alpha value `alpha` when drawing.
+
+The two main graphical primitives that are added are linear gradients and radial
+gradients:
```
Comment (anthrotype) [inline colr-gradients-spec.md:143]: this sentence that ends with a colon is referring to "gradients" but the section immediately following is titled "Color Line", I find that a bit confusing.

Comment (drott) [reply colr-gradients-spec.md]: Move this up to the beginning of the section and rephrase, PTAL.

Comment (anthrotype) [reply colr-gradients-spec.md]: looks good thanks

**Review by rsheeter — APPROVED** [review]

**Comment by rsheeter** [comment]
LGTM w/nits

**Pull request merged**